hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -282,45 +285,50 @@ private HoodieLogBlock createCorruptBlock(long blockStartPos) throws IOException
   }
 
   private boolean isBlockCorrupted(int blocksize) throws IOException {
-    long currentPos = inputStream.getPos();
-    long blockSizeFromFooter;
+    if (!StorageSchemes.isWriteTransactional(fs.getScheme())) {
```

Comment:
Let's invert this conditional and return early instead to reduce amount of code we need to change